### PR TITLE
test: add a lot of unit tests to improve the overall code coverage

### PR DESCRIPTION
**Pull request by hamada147**
# Overview

add a lot of unit tests to improve the overall code coverage

Fixes #<issue number>

### My PR contains...
* [x] No code changes (changes to documentation, CI, metadata, etc.)
* [ ] Bug fixes (non-breaking change which fixes an issue)
* [ ] Improvements (misc. changes to existing features)
* [ ] Features (non-breaking change which adds functionality)

### My changes...
* [ ] are breaking changes
* [x] are not breaking changes
* [ ] If yes to above: I have updated the documentation accordingly

### Documentation
* [x] My changes do not require a change to the project documentation
* [ ] My changes require a change to the project documentation
* [ ] If yes to above: I have updated the documentation accordingly

### Tests
* [x] My changes can not or do not need to be tested
* [ ] My changes can and should be tested by unit and/or integration tests
* [ ] If yes to above: I have added tests to cover my changes
* [ ] If yes to above: I have taken care to cover edge cases in my tests

### CODE REVIEW

**Comment by atala-dev** [comment]
### Code Coverage
|File|Coverage [25.00%]|:x:|
|:-|:-:|:-:|
|[apollo/src/androidMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/92881f3dc10df85af9d93419464002a81f01d550/apollo%2Fsrc%2FandroidMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PrivateKey.kt)|25.00%|:x:|
|[apollo/src/appleMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/92881f3dc10df85af9d93419464002a81f01d550/apollo%2Fsrc%2FappleMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PrivateKey.kt)|25.00%|:x:|
|[apollo/src/commonMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/92881f3dc10df85af9d93419464002a81f01d550/apollo%2Fsrc%2FcommonMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PrivateKey.kt)|25.00%|:x:|
|[apollo/src/jsMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/92881f3dc10df85af9d93419464002a81f01d550/apollo%2Fsrc%2FjsMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PrivateKey.kt)|25.00%|:x:|
|[apollo/src/jvmMain/kotlin/io/iohk/atala/prism/apollo/utils/KMMX25519PrivateKey.kt](https://github.com/input-output-hk/atala-prism-apollo/blob/92881f3dc10df85af9d93419464002a81f01d550/apollo%2Fsrc%2FjvmMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fapollo%2Futils%2FKMMX25519PrivateKey.kt)|25.00%|:x:|

|Total Project Coverage|95.20%|:white_check_mark:|
|:-|:-:|:-:|

File: apollo/src/commonTest/kotlin/io/iohk/atala/prism/apollo/base64/Base64Tests.kt [hunk 1]
```diff
@@ -191,4 +192,78 @@ class Base64Tests {
...
+
+    @Test
+    fun testStringBase64Encoded() {
+        assertEquals("aGVsbG8", "hello".base64Encoded)
+    }
+
+    @Test
+    fun testStringBase64Decoded() {
+        assertEquals("hello", "aGVsbG8".base64Decoded)
+    }
```
Comment (goncalo-frade-iohk) [inline apollo/src/commonTest/kotlin/io/iohk/atala/prism/apollo/base64/Base64Tests.kt:204]: We need to be much more specific with this tests, since this is a very common logic. When testing something like this, we need to search for the RFC and any test vectors or example cases, for example like the one below and the test name itself needs to reference the RFC and section.

https://datatracker.ietf.org/doc/html/rfc4648#page-12

If this doesn't exist then it should in the best case target a well know library test cases and test with their examples ;) this is as well for the tests written below.

File: apollo/src/commonTest/kotlin/io/iohk/atala/prism/apollo/derivation/DerivationAxisTests.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        val axis1 = DerivationAxis.normal(num1)
+        val axisHardened1 = DerivationAxis.hardened(num2)
+        val axis2 = DerivationAxis.normal(num2)
+
+        assertEquals(axis1, axis1)
+        assertNotEquals(axis1, axisHardened1)
+        assertNotEquals(axis1, axis2)
+    }
+}
```
Comment (goncalo-frade-iohk) [inline apollo/src/commonTest/kotlin/io/iohk/atala/prism/apollo/derivation/DerivationAxisTests.kt:81]: Ok now with this tests, HDKeyTests,  DerivationPathTests the same applies that ive said before.
Yes its great to add tests but we cannot just test for the sake of coverage ;), we need to add tests that are specifically targeting a test vector on a RFC or in this case a BIP.

BIP 32 specifies quite a few test vectors for Derivation paths and Hardened keys. This in part will make Apollo much more secure and reliable having those tests.

https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki

File: apollo/src/commonTest/kotlin/io/iohk/atala/prism/apollo/base64/Base64Tests.kt (hunk 1 above)
Comment (hamada147) [reply apollo/src/commonTest/kotlin/io/iohk/atala/prism/apollo/base64/Base64Tests.kt:204]: https://datatracker.ietf.org/doc/html/rfc4648#page-12 => We already have these tests implemented before.
These Base64 tests are related to the String and ByteArray extensions usage.

File: apollo/src/commonTest/kotlin/io/iohk/atala/prism/apollo/derivation/DerivationAxisTests.kt (hunk 2 above)
Comment (hamada147) [reply apollo/src/commonTest/kotlin/io/iohk/atala/prism/apollo/derivation/DerivationAxisTests.kt:81]: Oh nice, I will add these tests.
